### PR TITLE
[SPARK-38741][SQL][TESTS] Test the error class: MAP_KEY_DOES_NOT_EXIST

### PR DESCRIPTION
**Pull request by panbingkun**
## What changes were proposed in this pull request?
This PR aims to add a test for the error class MAP_KEY_DOES_NOT_EXIST to `QueryExecutionErrorsSuite`.

### Why are the changes needed?
The changes improve test coverage, and document expected error messages in tests.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
By running new test:
```
$ build/sbt "sql/testOnly *QueryExecutionAnsiErrorsSuite"
```

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 1]
```diff
@@ -400,4 +400,37 @@ class QueryExecutionErrorsSuite extends QueryTest
...
+      }
+      assert(e1.getErrorClass === "MAP_KEY_DOES_NOT_EXIST_IN_ELEMENT_AT")
+      assert(e1.getMessage ===
+        "Key 3 does not exist. To return NULL instead, use 'try_element_at'. " +
+          "If necessary set spark.sql.ansi.enabled to false to bypass this error." +
+          """
+            |== SQL(line 1, position 7) ==
+            |select element_at(map(1, 'a', 2, 'b'), 3)
+            |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            |""".stripMargin)
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:417]: Not need to indent additionally.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 2]
```diff
@@ -400,4 +400,37 @@ class QueryExecutionErrorsSuite extends QueryTest
...
+    }
+  }
+
+  test("MAP_KEY_DOES_NOT_EXIST: key does not exist") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:421]: Where is the checks of error class and error message?

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Comment by MaxGekk** [comment]
@panbingkun Could you resolve conflicts, here.

**Comment by MaxGekk** [comment]
@panbingkun ping

**Comment by panbingkun** [comment]
> @panbingkun ping

Done @MaxGekk

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**